theano/tensor/subtensor.py

```diff
@@ -2001,7 +2001,7 @@ def perform(self, node, inp, out_):
         if self.set_instead_of_inc:
             x[idx] = y
         else:
-            if config.cxx:
+            if config.cxx and node.inputs[0].dtype != 'float16':
```

Comment:
I think we should do an issue about that. It could be a but in numpy. Maybe in PyArray_BroadcastToShape. It could also depend of the numpy version.

which numpy version caused you this problem?